src/ios/BrotherPrinter.m

```diff
@@ -510,6 +534,45 @@ -(void)printViaSDK:(CDVInvokedUrlCommand*)command {
 }
 
 
+#pragma mark - Helpers
+
+-(CDVPluginResult *)resultForType:(NSString *)printerType {
+    CDVPluginResult *result;
+    
+    // Setup variables for retry
+    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
+    int maxNumberOfRetries = (int)[self integerValueFromDefaults:userDefaults forKey:kMaxNumberOfRetries withFallback:kDefaultMaxNumberOfRetries]; // Item 33
+    int retryTimeInSeconds = (double)[self integerValueFromDefaults:userDefaults forKey:kRetryTimeInSeconds withFallback:kDefaultRetryTimeInSeconds]; // Item 34
+
+    // if there is an error, try to requeue after X seconds
+    if (numberOfRetries < maxNumberOfRetries) {
+        [NSTimer scheduledTimerWithTimeInterval:retryTimeInSeconds
+                                            target:self
+                                        selector:@selector(retryPrint)
+                                        userInfo:nil
+                                        repeats:NO];
+        numberOfRetries++;
+        result = [self errorResult:printerType withCode:1 withMessage:[NSString stringWithFormat:@"unable to connect with device. Retrying in %i seconds (retries left: %i)", retryTimeInSeconds, maxNumberOfRetries-numberOfRetries]];
+    } else {
+        // else, if max number of retries reached, send error and reset retries variable
+        result = [self errorResult:printerType withCode:1 withMessage:@"unable to connect with device."];
```

Comment:
I don't know if we have any localization in place for these sort of error messages, but in case we do, it might be better if we send a message to the cordova app in the same way we do other error messages, where they are a simple string with an error code that gets interpretted by the cordova app...  But I think the way we are currently using these are standardized Brother error codes.

See `printService.js` and `BrotherPrintErrorService.js` on the kiosk app for reference.

If we do decide to use our error service, we will need to think of a way to handle the remaining number of retries in the messaging

Reply:
@dkohlruss I talked to Nem, and currently there are zero localization for any printer errors. If we are going to do this, it's going to be in another PR.